js_client/src/cobotVariableSelection.ts

```diff
@@ -1,51 +1,60 @@
+    { name: "digital_out1", readCommand: "get_digital_out(1)" },
+    { name: "digital_out2", readCommand: "get_digital_out(2)" },
+    { name: "digital_out3", readCommand: "get_digital_out(3)" },
+    { name: "digital_out4", readCommand: "get_digital_out(4)" },
+    { name: "digital_out5", readCommand: "get_digital_out(5)" },
+    { name: "digital_out6", readCommand: "get_digital_out(6)" },
+    { name: "digital_out7", readCommand: "get_digital_out(7)" },
+];
+
+export let selectedGlobalVariables = [...GLOBAL_DEBUG_VARIABLES];
+
+export function populateSelectGlobalVariableMenu(): void {
+    const dropdownMenu = document.getElementById("stateVariableSelection");
+    if (!dropdownMenu) return;
+
+    dropdownMenu.innerHTML = "";
+
+    GLOBAL_DEBUG_VARIABLES.forEach((variable) => {
+        const listItem = document.createElement("li");
+        const checkboxDiv = document.createElement("div");
+        checkboxDiv.classList.add("form-check", "dropdown-item");
+
+        const checkbox = document.createElement("input");
+        checkbox.classList.add("form-check-input");
+        checkbox.type = "checkbox";
+        checkbox.id = `checkbox-${variable.name}`;
+        checkbox.checked = true;
+        checkbox.value = variable.readCommand;
+
+        const label = document.createElement("label");
+        label.classList.add("form-check-label");
+        label.htmlFor = `checkbox-${variable.name}`;
+        label.textContent = `${variable.name}`;
+
+        checkbox.addEventListener("change", () => {
+            if (checkbox.checked) {
+                selectedGlobalVariables.push(variable);
+            } else {
+                selectedGlobalVariables = selectedGlobalVariables.filter((v) => v.name !== variable.name);
+            }
+        });
+
+        checkboxDiv.appendChild(checkbox);
+        checkboxDiv.appendChild(label);
+        listItem.appendChild(checkboxDiv);
+        dropdownMenu.appendChild(listItem);
     });
-
-    stateVariableSelectionList.replaceChildren(fragment);
-}
-
-function getListOfCheckedVariables(listOfNode: NodeListOf<HTMLInputElement>): string[] {
-    const checkedVariables: string[] = [];
-    listOfNode.forEach((checkbox: HTMLInputElement): void => {
-        if (checkbox.checked) {
-            checkedVariables.push(checkbox.id);
-        }
-    });
-    return checkedVariables;
-}
-
-export function listOfVariablesToDisplay(): string[] {
-    const checkboxes: NodeListOf<HTMLInputElement> = stateVariableSelectionList.querySelectorAll('.dropdown-item input[type="checkbox"]');
-    return getListOfCheckedVariables(checkboxes);
-
 }
 
-function isVariableSelectionGenerated(): boolean {
-    return !stateVariableSelectionList.firstElementChild.isSameNode(stateVariableSelectionList.lastElementChild);
-}
\ No newline at end of file
+// Populate the dropdown menu when the DOM is fully loaded
+document.addEventListener("DOMContentLoaded", populateSelectGlobalVariableMenu);
```

Comment:
Would it be possible to also listen on when a new inspection point have run, such that you can select the non global variables as well? I do not understand the code enough to see if this is possible